### DISCUSSION TITLE
Can not define MONGODB_URI

### DISCUSSION THREAD

**Discussion by Mark-Krensel** [neuefische/questions#279] [Web] [answered]
Hello, I want to connect may app to mongo db, but MONGODB_URI seems to be undefined.
I'm using the dbConnect.js (copy & pasted) and created a .env.local file with the connection string. 
As soon as I'm importing the (unused) dbConnect function it throws the error.

Error: ```Error: Please define the MONGODB_URI environment variable inside .env.local```

### /lib/dbConnect.js
```jsx
import mongoose from "mongoose";

const MONGODB_URI = process.env.MONGODB_URI;

if (!MONGODB_URI) {
  throw new Error(
    "Please define the MONGODB_URI environment variable inside .env.local"
  );
}

/**
 * Global is used here to maintain a cached connection across hot reloads
 * in development. This prevents connections growing exponentially
 * during API Route usage.
 */
let cached = global.mongoose;

if (!cached) {
  cached = global.mongoose = { conn: null, promise: null };
}

async function dbConnect() {
  if (cached.conn) {
    return cached.conn;
  }

  if (!cached.promise) {
    const opts = {
      bufferCommands: false,
    };

    cached.promise = mongoose.connect(MONGODB_URI, opts).then((mongoose) => {
      return mongoose;
    });
  }

  try {
    cached.conn = await cached.promise;
  } catch (e) {
    cached.promise = null;
    throw e;
  }

  return cached.conn;
}

export default dbConnect;
```

### .env.local
```
MONGODB_URI=mongodb+srv://M0ng0User:<password>@neuefische.qasmxiz.mongodb.net/<collection>?retryWrites=true&w=majority
```
Of course I replaced ```<password>``` & ```<collection>```.

**Comment by ThomasF85** [comment]
Hi Mark,

please make sure the .env.local file is in the root folder of your project.
Can you post a link to your current branch / pull request so I can reproduce the error?
One more thing:  Where you placed the collection name is wrong. This place in the connection string is for the database name. You can define the collection name in the mongoose Model.

**Comment by Mark-Krensel** [comment]
This is my active branch: https://github.com/Mark-Krensel/capstone-project/tree/feature/db
the .env file is inside the root folder. I already found out, that it may be read, as when I declare other variables I'm able to read those.
I guess I meant database instead of collection then :)

**Comment by ThomasF85** [comment, marked as answer]
Hi Mark,

the problem is that you try to access the database (and therefore the environment variables defined in .env.local) in the browser. You are trying to use the following function inside your weight page: `import { getAllDays } from "../services/dayService";`. This function internally calls dbConnect, which is trying to read the environment variables - which are not present in the browser for good reasons.
You can only call functions that use environment variables in server side code, for example: in your API functions, inside getServerSideProps, inside getStaticProps.

You have to use one of those mechanisms to get your data into the page.
